File: Drawflow.Server/Data/Models/FormTemplateInput.cs
namespace Drawflow.Server.Data.Models;

using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

#nullable disable

public class FormTemplateInput : Auditable
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long Id { get; set; }

    [Required]
    [ForeignKey("FormTemplate")]
    public long FormTemplateId { get; set; }
    public virtual FormTemplate FormTemplate { get; set; }

    public string Key { get; set; }

    public HtmlDataType HtmlDataType { get; set; }

    public JavaDataType JavaDataType { get; set; }

    public virtual ICollection<FormTemplateInputOption> Option { get; set; }
    public virtual ICollection<FormTemplateInputValidation> Validations { get; set; }
}

File: drawflow.client/src/app/features/forms/models/formTemplateInput.ts
import { Auditable } from '../../../shared/models/auditable';
import { FormTemplateInputOption } from './formTemplateInputOption';
import { FormTemplateInputValidation } from './formTemplateInputValidation';
import { HtmlDataType } from './htmlDataType';
import { JavaDataType } from './javaDataType';

export interface FormTemplateInput extends Auditable {
  id: number;
  formTemplateId: number;
  key: string;
  htmlDataType: HtmlDataType;
  javaDataType: JavaDataType;
  options: FormTemplateInputOption[];
  validations: FormTemplateInputValidation[];
}
